proxy.ts

```diff
@@ -40,4 +23,43 @@ export function getProxyUrl(reqUrl: url.Url): url.Url {
     }
 
     return proxyUrl;
+}
+
+
+export function checkBypass(reqUrl: url.Url): boolean {
+    if (!reqUrl.hostname) {
+        return false
+    }
+
+    let noProxy = process.env["no_proxy"] || process.env["NO_PROXY"] || '';
+    if (!noProxy) {
+        return false
+    }
+
+    // Determine the request port
+    let reqPort: number
+    if (reqUrl.port) {
+        reqPort = Number(reqUrl.port)
+    }
+    else if (reqUrl.protocol === 'http:') {
+        reqPort = 80
+    }
+    else if (reqUrl.protocol === 'https:') {
+        reqPort = 443
+    }
+
+    // Format the request hostname and hostname with port
+    let upperReqHosts = [reqUrl.hostname.toUpperCase()]
+    if (typeof reqPort === 'number') {
+        upperReqHosts.push(`${upperReqHosts[0]}:${reqPort}`)
+    }
+
+    // Compare request host against noproxy
```

Comment:
also note i check uppercase rather than lowercase which i believe is supposed to be better b/c of Turkish i